HopsanGUI/Widgets/LibraryWidget.cpp

```diff
@@ -650,231 +515,199 @@ void LibraryWidget::handleItemClick(QTreeWidgetItem *item, int column)
-                {
-                    gpModelHandler->saveState();
-                    ComponentLibraryEntry le = gpLibraryHandler->getEntry(typeNames.first());
-                    if (le.pLibrary)
-                    {
-                        // First unload the library
-                        QString libPath = le.pLibrary->xmlFilePath;
-                        bool rc = gpLibraryHandler->unloadLibraryByComponentType(typeNames.first());
-                        // NOTE! Now le and le.pLibrary is useless the pointer is dangling
-                        if (rc)
-                        {
-                            // We use the core generator directly to avoid calling the save state code in the library handler it does not seem to be working so well
-                            // But since we only need to unload one particular library this should work
-                            //! @todo fix the problem with save state
-                            auto spGenerator = createDefaultImportGenerator();
-                            bool compiledOK = spGenerator->compileComponentLibrary(libPath);
-                            if (!compiledOK)
-                            {
-                                gpMessageHandler->addErrorMessage("Library compiler failed");
-                            }
-
-                            // Now reload the library
-                            gpLibraryHandler->loadLibrary(libPath);
-                        }
-                    }
-                    gpModelHandler->restoreState();
-                }
-                // Handle check consistency
-                else if ((pReply == pCheckConsistenceAction) && !typeNames.isEmpty())
-                {
-                    ComponentLibraryEntry le = gpLibraryHandler->getEntry(typeNames.first());
-                    if (le.pLibrary)
-                    {
-                        auto spGenerator = createDefaultGenerator(false);
-                        bool checkOK = spGenerator->checkComponentLibrary(le.pLibrary->xmlFilePath);
-                        if (!checkOK) {
-                            gpMessageHandler->addWarningMessage(QString("The library '%1' has inconsistent component registration, this may cause exported models to fail.").arg(le.pLibrary->xmlFilePath));
-                        }
-                    }
-                }
+            else {
+                path = gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(pFirstSubComponentItem).value())->getBasePath();
             }
-            else if(pReply == pOpenFolderAction)
-            {
-                QDesktopServices::openUrl(QUrl("file:///" + gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(pFirstSubComponentItem).value())->getBasePath()));
+            QDesktopServices::openUrl(QUrl("file:///" + path));
+        }
+        else if(pReply == pEditXMLAction) {
+            if(!isComponentItem(item)) {
```

Comment:
I would like to have a isComponentLibraryItem help function here, for clarity

Reply:
How do you mean? Just rename the `isComponentItem`function?

Reply:
No, in this case it checks if the item is not a component then it must be a library. It would be easier to read if the code said 'if isLibrary'.
But I realised now that this also works on sub directories. So maybe not an.issue after all. But its a bit strange to be able to edit XML for a sub-directory, but not such a big deal